components/org.wso2.carbon.identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/endpoint/impl/ResourceRegistrationApiServiceImpl.java

```diff
@@ -203,81 +237,179 @@ public Response updateResource(String resourceId, ResourceDetailsDTO updatedReso
+                status = Response.Status.fromStatusCode(Integer.parseInt(statusCode));
+                errorCode = HandleErrorResponseConstants.MAP.get(code)[1];
+                isStatusOnly = false;
+            }
         }
-        return response;
+        throw buildResourceEndpointException(status, errorCode, umaException.getMessage(), isStatusOnly);
+    }
 
+    public ResourceEndpointException buildResourceEndpointException(Response.Status status, String errorCode,
+                                                                    String description, boolean isStatusOnly) {
+
+        if (isStatusOnly) {
+            return new ResourceEndpointException(status);
+        } else {
+            return new ResourceEndpointException(status, getErrorDTO(errorCode, description));
+        }
     }
 
     /**
-     * Handle error responses.
+     * Validate resourceId
      *
-     * @param umaException exception that should pass to handle error.
+     * @param resourceId resourceId of the resource which need to get deleted
+     * @return boolean
      */
-    public static void handleErrorResponse(UMAException umaException, Log log) throws ResourceEndpointException {
+    private static final boolean isResourceIdValid(String resourceId) throws UMAClientException {
 
-        HandleErrorResponseConstants handleErrorResponseConstants = new HandleErrorResponseConstants();
-        if (handleErrorResponseConstants.getResponseMap().containsKey(umaException.getErrorCode())) {
-            String statusCode = handleErrorResponseConstants.getResponseMap().get(umaException.getErrorCode())[0];
-            Response.Status status = Response.Status.fromStatusCode(Integer.parseInt(statusCode));
-            String errorCode = handleErrorResponseConstants.getResponseMap().get(umaException.getErrorCode())[1];
+        if (resourceId == null || StringUtils.isEmpty(resourceId)) {
+            if (log.isDebugEnabled()) {
+                log.debug("ResourceId is invalid");
+            }
+            throw new UMAClientException(ResourceConstants.ErrorMessages.ERROR_CODE_INVALID_RESOURCE_ID);
+        } else {
+            return true;
+        }
+    }
 
-            throw buildResourceEndpointException(status, errorCode, umaException.getMessage());
+    /**
+     * Retrieve resource owner name
+     *
+     * @param context message context
+     * @return resource owner name
+     */
+    private String getResourceOwner(MessageContext context) {
+
+        String resourceOwnerName = null;
+        if (context.getHttpServletRequest().getAttribute(AUTH_CONTEXT) instanceof AuthenticationContext) {
+            AuthenticationContext authContext = (AuthenticationContext) context.getHttpServletRequest()
+                    .getAttribute(AUTH_CONTEXT);
+            if (authContext.getUser() != null) {
+                return authContext.getUser().getUserName();
+            } else {
+                return null;
```

Comment:
What is the purpose of returning null here? If this is not returned, it will anyway return resourceOwnerName, which is null always.
resourceOwnerName is always null. Unless you assign the output of authContext.getUser().getUserName() to resourceOwnerName and return in the end, there is no point of having resourceOwnerName.